pkg/dns/aws/dns.go

```diff
@@ -157,6 +155,14 @@ func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error
 		}
 	}
 
+	// When RoleARN is provided, make a copy of the Route 53 session and configure it to use RoleARN.
+	// RoleARN is intended to only provide access to another account's Route 53 service, not for ELBs.
+	sessRoute53 := sess
+	if config.RoleARN != "" {
+		sessRoute53 = sess.Copy()
```

Comment:
@Miciah - I made a silly mistake that broke everything. I had `sessRoute53 := sess.Copy()` instead of `sessRoute53 = sess.Copy()`.

The variable was shadowed in the scope of the if statement, and didn't modify the `sessRoute53` in the scope we needed. Everything works as expected now.